besu/src/main/java/org/hyperledger/besu/services/BlockchainServiceImpl.java

```diff
@@ -50,13 +48,12 @@ public BlockchainServiceImpl() {}
   /**
    * Initialize the Blockchain service.
    *
-   * @param protocolContext the protocol context
+   * @param blockchain the blockchain
    * @param protocolSchedule the protocol schedule
    */
-  public void init(final ProtocolContext protocolContext, final ProtocolSchedule protocolSchedule) {
-    this.protocolContext = protocolContext;
+  public void init(final MutableBlockchain blockchain, final ProtocolSchedule protocolSchedule) {
```

Comment:
Any reason we are specifically requiring a MutableBlockchain and not just a Blockchain? It looks like all of the methods used below are present in Blockchain

Reply:
Sadly, this service does depend on the more specific MutableBlockchain in later, unmodified methods like storeBlock(), setFinalizedBlock() and setSafeBlock().